### PR TITLE
Update of the Entities section #285

### PR DESCRIPTION
**Pull request by dariavladykina**
This PR aims to update the Entities section as discussed in #285.

### CODE REVIEW

**Review by taroth21 — CHANGES_REQUESTED** [review]
@dariavladykina: Many thanks for the update! 

I'm sorry it took much longer to review (and more comments) than I originally expected because the devil is in the details, as usual.

That we currently have two different setups for entity files (the new one via doc kit and that some repositories still use the old setup where all entities are defined and maintained within the respective repository) does not make it easier for the explanations in the style guide....

Let me know in case of questions (or if you need help to find examples).

File: xml/docu_styleguide.manage.xml [hunk 1]
```diff
@@ -118,13 +118,121 @@
...
+      <term>Entity selection</term>
+      <listitem>
+       <para>
+        Use the entity name
+        <tag class="genentity">product</tag>
+        to identify the product for which the documentation is built. Set the
+        value of this entity once per release and have it expand to the name
+        of the current product:
+       </para>
+ <screen>&lt;phrase role="productname"&gt;&amp;product;&lt;/phrase&gt; includes 389-ds.</screen>
```
[the remark this answers is not in the document]
Comment (dariavladykina) [reply xml/docu_styleguide.manage.xml:188]: This was my main question - whether to keep this syntax or replace it with the entities. Thank you for clarifying!

File: xml/docu_styleguide.manage.xml [hunk 2]
```diff
@@ -118,13 +118,121 @@
...
+       <para>
+        If you need to reference a specific subproduct or a different product,
+        use a more specific entity:
+       </para>
+ <screen>The Ext4 file system has been included in the &lt;literal&gt;&amp;suse;&lt;/literal&gt;
+ Kernel since &lt;literal&gt;&amp;sle;&lt;/literal&gt; XYZ.</screen>
```
[the remark this answers is not in the document]
Comment (dariavladykina) [reply xml/docu_styleguide.manage.xml:194]: I believe this was the practice of working with entities in the past. Thanks for the new example!

File: xml/docu_styleguide.manage.xml [hunk 3]
```diff
@@ -195,107 +303,7 @@
      </callout>
     </calloutlist>
    </example>
-   <para>If an entity is placed at the beginning of a phrase or title, its resolved
```
[the remark this answers is not in the document]
Comment (dariavladykina) [reply xml/docu_styleguide.manage.xml:198]: This is a valid point, however, adding it here would demand changing this subsection's structure. Let's rather leave it under 'Acronyms' for now.

File: xml/docu_styleguide.manage.xml [hunk 4]
```diff
@@ -118,13 +118,128 @@
...
   <sect3 xml:id="sec-use-entity">
    <title>Using entity files</title>
-   <para>
-    &suse; uses a set of custom entities.
-    Find these entities in the file
-    <filename>xml/entity-decl.ent</filename> in each documentation repository.
-   </para>
+   <para>&suse; uses a set of custom entities.
+   Find these entities in the file <filename>generic-entities.ent</filename>
+   or <filename>entity-decl.ent</filename> in each documentation repository.</para>
```
Comment (taroth21) [inline xml/docu_styleguide.manage.xml:242]:
```suggestion
   Find these entities in the <filename>*.ent</filename>
   files in each documentation repository. One entity file can include other entity files.</para>
```

File: xml/docu_styleguide.manage.xml [hunk 5]
```diff
@@ -118,13 +118,121 @@
...
```
[the remark this answers is not in the document]
Comment (taroth21) [reply xml/docu_styleguide.manage.xml:242]: Maybe we can generalize as suggested below (by pointing to `*.ent` files).

File: xml/docu_styleguide.manage.xml [hunk 6]
```diff
@@ -153,14 +268,14 @@
...
+<screen>&lt;!ENTITY % entities SYSTEM "generic-entities.ent"&gt;
 %entities;</screen>
    <example xml:id="ex-entity">
-    <title>Excerpt from an <filename>entity-decl.ent</filename></title>
-<screen>&lt;!ENTITY<co xml:id="co-declaration"/> server<co xml:id="co-definition"/> "sun<co xml:id="co-value"/>"&gt;
-&lt;!ENTITY domain  "example.org"&gt;
-&lt;!ENTITY host    "&amp;server;.&amp;domain;<co xml:id="co-entity"/>"&gt;
-&lt;!ENTITY hosttag "&lt;uri xmlns='http://docbook.org/ns/docbook'&gt;&amp;host;&lt;/uri&gt;<co xml:id="co-tag"/>"&gt;</screen>
+    <title>Excerpt from <filename>product-entities.ent</filename></title>
+<screen>&lt;!ENTITY<co xml:id="co-declaration"/>product-sp<co xml:id="co-definition"/> "Code stream<co xml:id="co-value"/>"&gt;
```
Comment (taroth21) [inline xml/docu_styleguide.manage.xml:275]:
```suggestion
<screen>&lt;!ENTITY<co xml:id="co-declaration"/>product-sp<co xml:id="co-definition"/> "1<co xml:id="co-value"/>"&gt;
```

Comment (taroth21) [reply xml/docu_styleguide.manage.xml:275]: I would use a concrete example of a service pack number here to make it clearer.

File: xml/docu_styleguide.manage.xml [hunk 7]
```diff
@@ -153,14 +268,14 @@
...
+&lt;!ENTITY product-ga  "15"&gt;
+&lt;!ENTITY productnumber-regurl    "&amp;product-ga;.&amp;product-sp;<co xml:id="co-entity"/>"&gt;
+&lt;!ENTITY productnumber "&lt;phrase xmlns='http://docbook.org/ns/docbook' role='productnumber'&gt;&lt;/phrase"&gt;<co xml:id="co-tag"/>"&gt;</screen>
```
Comment (taroth21) [inline xml/docu_styleguide.manage.xml:278]:
```suggestion
&lt;!ENTITY productnumber "&lt;phrase xmlns='http://docbook.org/ns/docbook' role='productnumber'&gt;15&lt;/phrase"&gt;<co xml:id="co-tag"/>"&gt;</screen>
```

Comment (taroth21) [reply xml/docu_styleguide.manage.xml:278]: An entity value was still missing, I added `15` as a concrete example for a code stream number here.

File: xml/docu_styleguide.manage.xml [hunk 8]
```diff
@@ -135,11 +250,9 @@
     </listitem>
     <listitem>
      <para>
-      If you need to define new custom entities, add their new definitions
-      to the <filename>xml/entity-decl.ent</filename> file.
-      Do not include custom entity definitions directly in the file header,
-      unless the custom definitions are needed to override externally
-      embedded entities.
+      If you need a new entity to be used generically across all repositories, open a pull request against <filename>generic-entities.ent</fielname> in the Doc Kit repository. After the change has been approved by the Doc Kit maintainers, the entity update for <filename>generic-entities.ent</filename> will be rolled out to all Doc Kit-based repositories with the next Doc Kit run.
```
Comment (dariavladykina) [inline xml/docu_styleguide.manage.xml:253]:
```suggestion
      If you need a new entity to be used generically across all repositories, open a pull request against <filename>generic-entities.ent</filename> in the Doc Kit repository. After the change is approved by the Doc Kit maintainers, the entity update for <filename>generic-entities.ent</filename> will be rolled out to all Doc Kit-based repositories with the next Doc Kit run.
```

File: xml/docu_styleguide.manage.xml [hunk 9]
```diff
@@ -135,11 +250,9 @@
...
+     If you need a custom entity that only applies to a specific repository, define it in <filename>product-entities.ent<filename> or in <filename>entities-decl.ent</filename> in this specific repository.</para>
+      <para>Do not include custom entity definitions directly in the file header, unless the custom definitions  are needed to override externally embedded entities.
```
Comment (dariavladykina) [inline xml/docu_styleguide.manage.xml:255]:
```suggestion
      <para>Do not include custom entity definitions directly in the file header, unless the custom definitions are needed to override externally embedded entities.
```

File: xml/docu_styleguide.manage.xml [hunk 10]
```diff
@@ -118,13 +118,121 @@
...
+    <title>Common types of entities</title>
+    <para>Official generic entities are maintained in the Doc Kit repository.
+      They include &suse; product names and other terms that are valid for every 
+      repository. The file <filename>generic-entities.ent</filename> must not 
+      be changed. If there is a need to change an entity, this can be done
+      in the <filename>product-entities.ent</filename> or 
+      <filename>entity-decl.ent</filename> file in the respective repository.</para>
+    <para>
+     A <filename>generic-entities.ent</filename> or 
+     <filename>entity-decl.ent</filename> file contains several categories of
```
[the remark this answers is not in the document]
Comment (janajaeger) [reply xml/docu_styleguide.manage.xml:137]: What Tanja said ;)

**Review by jfaltenbacher — APPROVED** [review]

**Pull request merged**